src/super_gradients/training/utils/sg_trainer_utils.py

```diff
@@ -190,16 +190,22 @@ def update_monitored_values_dict(monitored_values_dict: Dict[str, MonitoredValue
     :param new_values_dict: Dict mapping value names to their new (i.e. current epoch) value.
     :return: Updated monitored_values_dict
     """
-    for monitored_value_name in monitored_values_dict.keys():
+    relevant_keys = set(new_values_dict.keys()).intersection(monitored_values_dict.keys())
+    for monitored_value_name in relevant_keys:
+        previous_value = monitored_values_dict[monitored_value_name]
         monitored_values_dict[monitored_value_name] = update_monitored_value(
             new_value=new_values_dict[monitored_value_name],
-            previous_monitored_value=monitored_values_dict[monitored_value_name],
+            previous_monitored_value=previous_value,
         )
     return monitored_values_dict
 
 
 def display_epoch_summary(
-    epoch: int, n_digits: int, train_monitored_values: Dict[str, MonitoredValue], valid_monitored_values: Dict[str, MonitoredValue]
+    epoch: int,
+    n_digits: int,
+    train_monitored_values: Dict[str, MonitoredValue],
+    valid_monitored_values: Dict[str, MonitoredValue],
+    test_monitored_values: Dict[str, MonitoredValue],
```

Comment:
I feel uneasy about duplicated code. What if we play a little bit future-proof here?
Let's say we add a single parameter: `loader_metrics: Dict[str, Dict[str, MonitoredValue]]` instead of `train_monitored_values, valid_monitored_values, test_monitored_values`.

The call to the method would obviously change, but not that significantly:
```
display_epoch_summary(
  loader_metrics=dict(
    train=train_monitored_values,
    valid=valid_monitored_values,
    test=test_monitored_values
  )
}
```

However internal implementation would be simplified since we would get rid of duplicated code that generates the summary tree for each loader. Instead we could simply iterate over the collection.
If has another advantage that if we will get to the multiple validation loaders - this function will be already ready for this change.
WDYT?

Reply:
Well, I see two different things and am not sure which one you are thinking about:
1. Grouping all  the "splits" monitored values into a single dict: `Dict[str, Dict[str, MonitoredValue]]`
```python
loader_metrics=dict(
    train=train_monitored_values,
    valid=valid_monitored_values,
    test=test_monitored_values
  )
```
2. Allowing each split to support nested monitored values (1 per loader): `Dict[str, Dict[str, MonitoredValue]]`
```python
valid_monitored_values = {
    "dataset_1": dataset1_monitored_values, ...
    "dataset_2": dataset2_monitored_values, ...
}
```
Which one are you referring to?
I see less value in the 1. because we always work with train/test/valid, so I don't see a lot of value into abstracting the split away. I think that the solution would rather be to create a single class MonitoredValues (or a better name) that would handle train/valid/test and take out the code from the Trainer.
What do you think?

Reply:
I mean the first option)

Reply:
Like this? https://github.com/Deci-AI/super-gradients/pull/1241/commits/1d890dd8fe92d8d7643b2e9c155c51a9d89d37dc